### PR TITLE
✨ Dashboard Table Filters for Sanger VC and Mutect2 VC Status (#2018 #2019)

### PR DESCRIPTION
**Pull request by mistryrn**
**Description of changes**

* Connect dashboard table filters to API for `Sanger VC` and `Mutect2 Status` columns
* Refactor dashboard table filter options lists to reduce repeated code

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [ ] Modified Existing components and updates stories
  - [x] No new UIKit components or changes
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: components/pages/submission-system/program-dashboard/DonorDataSummary/common.tsx [hunk 1]
```diff
@@ -22,15 +22,15 @@ import { DonorDataReleaseState } from './types';
 import { DataTableStarIcon as StarIcon } from '../../common';
 
 export const RELEASED_STATE_FILL_COLOURS: {
-  [k in DonorDataReleaseState]: React.ComponentProps<typeof StarIcon>['fill']
+  [k in DonorDataReleaseState]: React.ComponentProps<typeof StarIcon>['fill'];
```
Comment (mistryrn) [inline components/pages/submission-system/program-dashboard/DonorDataSummary/common.tsx:25]: Just noticed this change and the one on line 33. I didn't make these, I'm assuming the linter added the semicolons here. Not sure if they make a difference, anyone know?

Comment (ciaranschutte) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/common.tsx:25]: it's prettier. builds and typechecks fine. no difference as far as I know. TS is usually a little stricter with these declarations.

**Review by ciaranschutte — APPROVED** [review]
Looks great! Love that filter. Nice work!

File: components/pages/submission-system/program-dashboard/DonorDataSummary/common.tsx (hunk 1 above)
Comment (mistryrn) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/common.tsx:25]: Ah, makes sense. Thanks for clearing that up! :)

**Pull request merged**